actionpack/lib/action_dispatch/middleware/ssl.rb

```diff
@@ -1,72 +1,136 @@
 module ActionDispatch
+  # This middleware is added to the stack when `config.force_ssl = true`.
+  # It does three jobs to enforce secure HTTP requests:
+  #
+  #   1. TLS redirect. http:// requests are permanently redirected to https://
+  #      with the same URL host, path, etc. Pass `:host` and/or `:port` to
+  #      modify the destination URL. This is always enabled.
+  #
+  #   2. Secure cookies. Sets the `secure` flag on cookies to tell browsers they
+  #      mustn't be sent along with http:// requests. This is always enabled.
+  #
+  #   3. HTTP Strict Transport Security (HSTS). Tells the browser to remember
+  #      this site as TLS-only and automatically redirect non-TLS requests.
+  #      Enabled by default. Pass `hsts: false` to disable.
+  #
+  # Configure HSTS with `hsts: { … }`:
+  #   * `expires`: How long, in seconds, these settings will stick. Defaults to
+  #     `18.weeks`, the minimum required to qualify for browser preload lists.
+  #   * `subdomains`: Set to `true` to tell the browser to apply these settings
+  #     to all subdomains. This protects your cookies from interception by a
+  #     vulnerable site on a subdomain. Defaults to `false`.
+  #   * `preload`: Advertise that this site may be included in browsers'
+  #     preloaded HSTS lists. HSTS protects your site on every visit *except the
+  #     first visit* since it hasn't seen your HSTS header yet. To close this
+  #     gap, browser vendors include a baked-in list of HSTS-enabled sites.
+  #     Go to https://hstspreload.appspot.com to submit your site for inclusion.
+  #
+  # Disabling HSTS: To turn off HSTS, omitting the header is not enough.
+  # Browsers will remember the original HSTS directive until it expires.
+  # Instead, use the header to tell browsers to expire HSTS immediately.
+  # Setting `hsts: false` is a shortcut for `hsts: { expires: 0 }`.
   class SSL
-    YEAR = 31536000
+    # Default to the minimum expiry needed to qualify for browser preload.
+    HSTS_EXPIRES_IN = 18.weeks
 
     def self.default_hsts_options
-      { :expires => YEAR, :subdomains => false }
+      { expires: HSTS_EXPIRES_IN, subdomains: false, preload: false }
     end
 
-    def initialize(app, options = {})
-      @app = app
-
-      @hsts = options.fetch(:hsts, {})
-      @hsts = {} if @hsts == true
-      @hsts = self.class.default_hsts_options.merge(@hsts) if @hsts
+    def initialize(app, redirect: {}, hsts: {}, **options)
+      if options[:host] || options[:port]
+        ActiveSupport::Deprecation.warn <<-end_warning.strip_heredoc
+          The `:host` and `:port` options are moving within `:redirect`:
+          `config.ssl_options = { redirect: { host: …, port: … }}`.
+        end_warning
+        redirect = options.slice(:host, :port)
+      end
 
-      @host    = options[:host]
-      @port    = options[:port]
+      @app = RedirectInsecureRequests.new(app, **redirect)
+      @hsts = normalize_hsts_options(hsts)
```

Comment:
I don't see `@hsts` being used anywhere else than in this scope. Would it make sense to just have it as a local variable?